com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/PollServiceOperationsExecution.java

```diff
@@ -116,20 +116,31 @@ public AsyncExecutionState execute(ExecutionWrapper execution) {
     private List<CloudServiceExtended> computeServicesToPoll(DelegateExecution context,
         Map<String, ServiceOperationType> triggeredServiceOperations) {
 
-        List<String> servicesToDeleteNames = StepsUtil.getServicesToDelete(context);
-        Map<String, String> serviceGuids = StepsUtil.getServicesGuids(context);
-
-        List<CloudServiceExtended> servicesToDelete = servicesToDeleteNames.stream()
-            .map(name -> new CloudServiceExtended(new Meta(UUID.fromString(serviceGuids.get(name)), null, null), name))
+        List<CloudServiceExtended> servicesToDelete = getServicesToDelete(context);
+        List<CloudServiceExtended> nonUserServicesToCreate = getServicesToCreate(context);
+        
+        return Stream.concat(nonUserServicesToCreate.stream(), servicesToDelete.stream())
+            .filter(service -> triggeredServiceOperations.containsKey(service.getName()))
             .collect(Collectors.toList());
-        List<CloudServiceExtended> servicesToCreate = StepsUtil.getServicesToCreate(context);
+    }
 
+    private List<CloudServiceExtended> getServicesToCreate(DelegateExecution context) {
+        List<CloudServiceExtended> allServicesToCreate = StepsUtil.getServicesToCreate(context);
         // There's no need to poll the creation or update of user-provided services, because it is done synchronously:
-        Stream<CloudServiceExtended> nonUserServicesToCreate = servicesToCreate.stream()
-            .filter(s -> !s.isUserProvided());
-        return Stream.concat(nonUserServicesToCreate, servicesToDelete.stream())
-            .filter(service -> triggeredServiceOperations.containsKey(service.getName()))
+        List<CloudServiceExtended> nonUserServicesToCreate = allServicesToCreate.stream()
+            .filter(s -> !s.isUserProvided())
+            .collect(Collectors.toList());
+        return nonUserServicesToCreate;
+    }
+
+    private List<CloudServiceExtended> getServicesToDelete(DelegateExecution context) {
+        List<String> servicesToDeleteNames = StepsUtil.getServicesToDelete(context);
+        Map<String, String> servicesGuids = StepsUtil.getServicesGuids(context);
+
+        List<CloudServiceExtended> servicesToDelete = servicesToDeleteNames.stream()
+            .map(name -> new CloudServiceExtended(new Meta(UUID.fromString(servicesGuids.get(name)), null, null), name))
```

Comment:
Why new Meta(UUID.fromString(servicesGuids.get(name))? Do you need this guid?
Could it be some random?

Reply:
We need it in the cloudServiceInstance